### PR TITLE
fixes amazon isbn lookup to work w/ isbn13

### LINKED ISSUE #1456: Fix ISBN API openlibrary.org/isbn/{xxx} to gracefully handle error cases
Today, when one hits the endpoint openlibrary.org/isbn/{xxx}, the following algorithm is performed:

1. try to fetch the OL edition for the isbn10/13 if it exists
2. If no OL edition exists, attempt lookup on amazon with isbn10 as `asin`
3. If it happens to be isbn13 (which doesn't match `asin` or if isbn10 not on amazon, fail miserably :feelsgood: 

The problem occurs when we hit (2) above.
a) if isbn10/13 does not match an Open Library edition, try to convert to isbn10 (because the amazon lookup API uses isbn10)
b) if `get_amazon_metadata` returns `None`, gracefully fail and return a valid and useful json error message instead of a server error

### PR DESCRIPTION
**Pull request by mekarpeles**
Closes #1456

### CODE REVIEW

File: openlibrary/core/vendors.py [hunk 1]
```diff
@@ -60,14 +59,14 @@ def _get_amazon_metadata(isbn):
         'languages': list(product.languages),  # needs to be normalized
         'publishers': [product.publisher],
         'cover': product.large_image_url,
-        'isbn_10': [isbn_10],
-        'isbn_13': [isbn_13]
+        'isbn_10': [isbn if len(isbn) == 10 else isbn_13_to_isbn_10(isbn)],
```
Comment (mekarpeles) [inline openlibrary/core/vendors.py]: this could break if the isbn13 has no isbn10 equivalent

Comment (LeadSongDog) [reply openlibrary/core/vendors.py]: @mekarpeles how is that possible? Don't all isbn10 values map to a bookland isbn13?

**Comment by mekarpeles** [comment]
reviewed by @hornc

**Comment by LeadSongDog** [comment]
@mekarpeles It seems that the process of computing isbn10 from isbn13 (or reverse) is being used to build the solr index, but that the two are not both being saved in the edition record. We continue to see one or the other on their own. Surely this should be an easy win to fix.

**Comment by mekarpeles** [comment]
I think I may agree -- @hornc I think would rather us store only isbn13 (?) and or specifically not store both

**Pull request merged**